### ISSUE TITLE
SHA256 bug from variable

### ISSUE THREAD

**Issue by ahpaleus** [hackvertor/hackvertor#44]
Hello,

I have found an annoying issue that the sha256 hash from a variable has different values depending on the place in the request.

As we know, SHA256("test1337") is `168ba29eebd5d1e904b4e0d318d164a9098774191a122aefb9df248db6fbdfb1`, but it gives a wrong value when we try to hash it before set_variable tag.

Example:
Input
```
POST /api/ HTTP/1.1
X-Test: <@sha256><@get_variable1/><@/sha256>
Content-Type: application/json

<@set_variable1>test1337<@/set_variable1>

<@sha256><@get_variable1/><@/sha256>
```

Output:
```
POST /api/ HTTP/1.1
X-Test: 54c9e4bd59ec271ab579718c946b022e36d4d9cf3a562ee35c70d4beff0d8335
Content-Type: application/json

test1337

168ba29eebd5d1e904b4e0d318d164a9098774191a122aefb9df248db6fbdfb1
```

It is not possible to nest a tag before setting a variable -> `54c9e4bd59ec271ab579718c946b022e36d4d9cf3a562ee35c70d4beff0d8335` is SHA256 of `<@get_variable1/>`. Without the SHA256 tag -> `<@get_variable1/>` gives a valid value (`test1337`, in this case).

**Comment by hackvertor** [comment]
Thanks for the bug report. This looks like a problem introduced with the new parser work. I think previously I used regexes and extracted the contents of set tags from outside to in which means you could store conversions inside set tags. I need to do this again with the new code. Sorry for this I'll get it fixed ASAP.

**Comment by robotate** [comment]
The clever "null" fix for #39 didn't account for nesting. Here are some tests I ran before I realized it was already reported, in case helpful.
![Screen Shot 2021-03-09 at 11 48 35 PM](https://user-images.githubusercontent.com/2855914/110595555-fd1fce00-8132-11eb-988c-5df9c436c43c.png)
